### PR TITLE
[Bug 1885852, 1885859]Resolve issues for IntelliJ 2021.3 EAP 

### PR DESCRIPTION
**Pull request by Flanker32**
- Fix logic to get module root [AB#1885859](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1885859) 
- Fix slow operation in EDT issues [AB#1886500](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1886500)
    - Function local run/deployment
    - Spark run configuration action (Refers https://github.com/JetBrains/intellij-community/blob/master/platform/core-api/src/com/intellij/util/SlowOperations.java#L34-L68) @wezhang @t-rufang Please help review the changes
- Fix invalid read access for azure artifact [AB#1885852](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1885852)

### CODE REVIEW

**Review by t-rufang — APPROVED** [review]
LGTM! Thanks Hanxiao

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java [hunk 1]
```diff
@@ -64,7 +65,7 @@ public void setArtifact(@Nullable final AzureArtifact azureArtifact) {
     )
     protected List<? extends AzureArtifact> loadItems() throws Exception {
         final List<AzureArtifact> collect = fileArtifactOnly ?
-                new ArrayList<>() : AzureArtifactManager.getInstance(project).getAllSupportedAzureArtifacts();
+                new ArrayList<>() : ReadAction.compute(() -> AzureArtifactManager.getInstance(project).getAllSupportedAzureArtifacts());
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java]: this fix can be reverted as I have fixed it at "root": https://github.com/microsoft/azure-tools-for-java/commit/ff32305bc9387ca29c1f805401ff4923d16a6f01

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-springcloud/src/main/java/com/microsoft/azure/toolkit/intellij/springcloud/deplolyment/SpringCloudDeploymentConfiguration.java [hunk 2]
```diff
@@ -69,7 +70,7 @@ public void readExternal(Element element) throws InvalidDataException {
                 .orElse(SpringCloudAppConfig.builder().deployment(SpringCloudDeploymentConfig.builder().build()).build());
         Optional.ofNullable(element.getChild("Artifact"))
                 .map(e -> e.getAttributeValue("identifier"))
-                .map(manager::getAzureArtifactById)
+                .map(id -> ReadAction.compute(() -> manager.getAzureArtifactById(id)))
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-springcloud/src/main/java/com/microsoft/azure/toolkit/intellij/springcloud/deplolyment/SpringCloudDeploymentConfiguration.java]: this fix can be reverted as I have fixed it at "root": https://github.com/microsoft/azure-tools-for-java/commit/ff32305bc9387ca29c1f805401ff4923d16a6f01

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/azure/toolkit/intellij/function/runner/core/FunctionUtils.java [hunk 3]
```diff
@@ -142,8 +143,7 @@ public static void cleanUpStagingFolder(File stagingFolder) {
                 return false;
             }
             final GlobalSearchScope scope = GlobalSearchScope.moduleWithLibrariesScope(m);
-            final PsiClass ecClass = JavaPsiFacade.getInstance(project).findClass(AZURE_FUNCTION_ANNOTATION_CLASS,
-                                                                                  scope);
+            final PsiClass ecClass = ReadAction.compute(() -> JavaPsiFacade.getInstance(project).findClass(AZURE_FUNCTION_ANNOTATION_CLASS, scope));
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/azure/toolkit/intellij/function/runner/core/FunctionUtils.java]: We can use `AzureTaskManager#read`, or `#readAsObservable` which handles errors automatically.

**Review by wangmingliang-ms — CHANGES_REQUESTED** [review]
please resolve my comments

**Review by wangmingliang-ms — APPROVED** [review]

**Pull request merged**